src/com/docker/projects/Documentation.groovy

```diff
@@ -0,0 +1,51 @@
+
+
+package com.docker.projects;
+
+// Add the following code to any Jenkinsfile.
+// It will return with success if there are no changes in the specified docs dir
+// and Will return with success if run on a non Pull Request branch
+
+// def docs = new com.docker.projects.Documentation()
+// docs.documentationPRChecker("docs")
+
+// docsDir ~~ the subdirectory within your repo that the documentation resides
+def documentationPRChecker(docsDir) {
+  if (env.CHANGE_ID != null) {
+    stage "docs PR checker"
+
+    tokens = "${env.JOB_NAME}".tokenize('/')
+    org = tokens[0]
+    repo = tokens[1]
+    branch = tokens[2]
+    sh "git log --format='%H' -1 > tempoutput"
+    sha = readFile("tempoutput").trim()
+    imageName = "${repo}:${branch}"
+    containerName = "${repo}-${branch}-${env.BUILD_ID}"
+
+    sh "env > env.txt" // in case I need to debug
+
+    sh "git log origin/master..${sha} ${docsDir} > tempoutput"
+    changes = readFile("tempoutput").trim()
+    if (changes.size() == 0) {
+      echo "no changes found in ${docsDir}"
+    } else {
+      echo changes
+      //slackSend channel: "#docs-automation", message: "Starting docs test of <a href='${env.CHANGE_URL}'>${imageName}</a> - see <a href='${env.BUILD_URL}console'>console</a>"
+      //setGitHubPullRequestStatus message: "starting docs test"
+      sh "docker pull docs/base:oss"
+      dir(docsDir) {
+        sh "docker build -t ${imageName} ."
+      }
+      sh "docker run --name=${containerName} ${imageName}"
+
+      // TODO: summarize the changes & errors (these are files used by GHPRB and the summary plugin
+      sh "docker cp ${containerName}:/validate.junit.xml ."
```

Comment:
probably want to archive the junit file somewhere in this function:

```groovy
step([$class: 'JUnitResultArchiver', testResults: 'validate.junit.xml', keepLongStdio: true])
```

FYI it will change `currentBuild.result` to `'UNSTABLE'` if there are errors. this is important to note if any code following this function call wants to bail out early instead of going on to do more work. i don't think that's curretnyl the case with the way that most of the docs builds work, but just in case.